_posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md

```diff
@@ -0,0 +1,71 @@
+---
+layout: post
+title:  "Haskell Learning Group: Our Progress and Updates"
+date: 2017-11-07 00:03:30 +0200
+categories: haskell how-we-learn
+author_name: Sergii Paryzhskyi
+author_url : /author/sergii_paryzhskyi
+author_avatar: sergii_paryzhskyi
+show_avatar : true
+read_time : 7
+show_related_posts: false
+square_related: recommend-wolf
+feature_image: posts/2017-10-17-first-haskell-meeting/haskell-poster.jpg
+---
+
+This post is a quick update and the latest news about our Haskell learning group. Even though there were no blog posts on this topic recently, we are still having weekly meetings and mostly stick to the plan. That of course implies preparation time, spend between our sessions. There is no rush in our learning process and we take time to discuss interesting topics a bit longer, especially when all participants are involved. That being said, we might have a plan to go through one chapter per session but it also fine if it is less than that.
+
+As expected the difficulty is slowly increasing as we go along. We are spending more time reading chapters. Sometimes the same material has to be read multiple times in order to fully understand what is it about.
+
+It helps a lot that the most of the concepts are familiar to us through other languages or libraries we are using. For instance JavaScript doesn't support currying natively the way Haskell does it, but there are libraries which introduce these concepts. We are very often using [RamdaJS][ramdajs] in our projects, so such things as function composition, currying, partial application and so on were not something totally new to us. They still had to be "translated" and demonstrated in a Haskell way but we certainly didn't need much time to get the idea.
+
+Similar situation was with typing system and type definitions. As many people from the group doing i.e. TypeScript or Scala, there were not that many surprises for us. Nevertheless it's pretty impressive what can be achieved with type constraints in Haskell.
```

Comment:
"Similar situation was with typing system" is a somewhat odd phrasing. 

Maybe try something like:

"The same applies to the the parts that we learned about the Haskell type system so far: ..."

I would be careful about not making too broad judgements. We have so far only talked only about the absolutely super tiny minimal first tidbits of the type system. The statement that "typing system and type definitions" are not surprising because people know Typescript and Scala might not be well received or sound silly. So it would probably help to restrict that statement a bit.

Reply:
typing system -> type system
"type constraints" is probably (?) not what you actually want to use? I think you want to refer to the type system in general?

Reply:
I've rewrote the sentence you've mentioned. 
As this blogpost meant to tell about our small steps in learning haskell, I think its fine to write about it from our perspective and base on what we know so far. I guess in next posts I can refer to this paragraph and say how wrong I was by thinking so :)

"type constraints" -> Constraints that are dictated by types, i.e. this argument must be an Integer.